ruby/lib/ci/queue/static.rb

```diff
@@ -64,9 +64,24 @@ def acknowledge(test)
         true
       end
 
+      def increment_test_failures
+        @test_failures = test_failures + 1
+      end
+
+      def test_failures
+        @test_failures ||= 0
+      end
+
+      def max_test_failures_reached?
+        return false if config.max_test_failures.nil?
+
+        test_failures >= config.max_test_failures
```

Comment:
I think there is a difference of meaning between "test_failures" and "failed_tests". I think the latter's meaning is more acute in the sense that it is counting the tests that failed, not counting the requeues.